src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/SingleBodyParameterMessageFormatter.cs

```diff
@@ -0,0 +1,387 @@
+                }
+            }
+            else
+            {
+                serializer = null;
+            }
+
+            return new SingleParameterBodyWriter(body, serializer);
+        }
+
+        protected virtual object ReadObject(Message message)
+        {
+            if (HttpStreamFormatter.IsEmptyMessage(message))
+            {
+                return null;
+            }
+
+            XmlObjectSerializer[] inputSerializers = GetInputSerializers();
+            XmlDictionaryReader reader = message.GetReaderAtBodyContents();
+            if (inputSerializers != null)
+            {
+                for (int i = 0; i < inputSerializers.Length; ++i)
+                {
+                    if (inputSerializers[i].IsStartObject(reader))
+                    {
+                        return inputSerializers[i].ReadObject(reader, false);
+                    }
+                }
+            }
+
+            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new SerializationException(SR.Format(SR.CannotDeserializeBody, reader.LocalName, reader.NamespaceURI, OperationName, ContractName, ContractNs, _serializerType)));
+        }
+
+        internal class NullMessageFormatter : IDispatchMessageFormatter, IClientMessageFormatter
+        {
+            // TODO: Can I remove this?
+
+            //readonly bool _useJson;
+            //readonly string _callbackParameterName;
+
+            //public NullMessageFormatter(bool useJson, string callbackParameterName)
+            //{
+            //    _useJson = useJson;
+            //    _callbackParameterName = callbackParameterName;
+            //}
+
+            public object DeserializeReply(Message message, object[] parameters)
+            {
+                return null;
+            }
+
+            public void DeserializeRequest(Message message, object[] parameters)
+            {
+            }
+
+            public Message SerializeReply(MessageVersion messageVersion, object[] parameters, object result)
+            {
+                Message reply = Message.CreateMessage(messageVersion, null);
+                SuppressReplyEntityBody(reply);
+                // TODO: Can I remove this?
+                //if (_useJson && WebHttpBehavior.TrySetupJavascriptCallback(_callbackParameterName) != null)
+                //{
+                //    reply.Properties.Add(WebBodyFormatMessageProperty.Name, WebBodyFormatMessageProperty.JsonProperty);
+                //}
+                return reply;
+            }
+
+            public Message SerializeRequest(MessageVersion messageVersion, object[] parameters)
+            {
+                Message request = Message.CreateMessage(messageVersion, null);
+                // TODO: Can I remove this?
+                //SuppressRequestEntityBody(request);
```

Comment:
Method should go away once removed client interface.

Reply:
This has been addressed.